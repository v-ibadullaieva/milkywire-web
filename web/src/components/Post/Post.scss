.LinksWrap {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.Link {
  font-size: 16px;
  font-weight: 600;
  padding: 5px;
  margin-bottom: 10px;

  &:hover {
    color: #afafaf;
    cursor: pointer;
  }
}