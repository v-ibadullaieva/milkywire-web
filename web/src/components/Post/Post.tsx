import React from 'react';
import { Card } from 'antd';

import './Post.scss';
import { PostType } from '../PostsList/PostsList';

interface PostProps extends React.Props<any> {
  post: PostType,
  handlePrev: () => void,
  handleNext: () => void
}

const Post = ({ post, handlePrev, handleNext }: PostProps) => {
  const { image, version, width, height } = post.data.media[0];

  return (
    <div>
      <div className='LinksWrap'>
        <span className='Link' onClick={handlePrev}>go to prev</span>
        <span className='Link' onClick={handleNext}>go to next</span>
      </div>
      <Card
        cover={<img alt='img' src={image} />}
      >
        <p>{post.description}</p>
        <p><b>version:</b> {version}</p>
        <p><b>size:</b> {width} x {height}</p>
      </Card>
    </div>
  )
}

export default Post;
