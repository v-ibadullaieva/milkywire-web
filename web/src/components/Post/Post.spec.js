import React from 'react';
import { create } from "react-test-renderer";

import Post from './Post';

const POST = {
  id: 0,
  type: "IMAGES",
  description: "lol",
  data: {
    media: [{
      image: "https://picsum.photos/id/0/5616/3744",
      width: 5616,
      height: 3744,
      version: "2019-03-14",
      description: "Alejandro Escamilla"
    }]
  },
  impacter_id: 0
};

describe('Post component', () => {
  test("Matches the snapshot", () => {
    const post = create(<Post post={POST} />);
    expect(post.toJSON()).toMatchSnapshot();
  });
});
