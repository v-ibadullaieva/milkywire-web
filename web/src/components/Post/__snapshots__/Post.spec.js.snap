// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component Matches the snapshot 1`] = `
<div>
  <div
    className="LinksWrap"
  >
    <span
      className="Link"
    >
      go to prev
    </span>
    <span
      className="Link"
    >
      go to next
    </span>
  </div>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-cover"
    >
      <img
        alt="img"
        src="https://picsum.photos/id/0/5616/3744"
      />
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <p>
        lol
      </p>
      <p>
        <b>
          version:
        </b>
         
        2019-03-14
      </p>
      <p>
        <b>
          size:
        </b>
         
        5616
         x 
        3744
      </p>
    </div>
  </div>
</div>
`;
